File: FizzBuzz/FSharp/FSharpFizzBuzz/UnitTestFSharp/FizzBuzzTest.fs
namespace UnitTestFSharp

open NUnit.Framework

module Say =
    let [<Test>] ``say 1 returns '1'``() =
       let expected = "1"
       Assert.AreEqual(expected, FizzBuzz.FizzBuzz.say (1))

    let [<Test>] ``say 3 returns 'Fizz'``() =
        let expected = "Fizz"
        Assert.AreEqual(expected, FizzBuzz.FizzBuzz.say (3))

    let [<Test>] ``say 6 returns 'Fizz'``() =
        let expected = "Fizz"
        Assert.AreEqual(expected, FizzBuzz.FizzBuzz.say (6))
        
    let [<Test>] ``say 5 returns 'Buzz'``() =
        let expected = "Buzz"
        Assert.AreEqual(expected, FizzBuzz.FizzBuzz.say (5))
    
    let [<Test>] ``say 15 returns 'FizzBuzz'``() =
        let expected = "FizzBuzz"
        Assert.AreEqual(expected, FizzBuzz.FizzBuzz.say (15))

File: fsharp/FizzBuzz/JavaScript/20190926-FizzBuzz-ECMAScript6/test/FizzBuzz_test.js
import {expect} from 'chai'
import FizzBuzz from "../app/FizzBuzz"

describe("FizzBuzz", () => {
  it("1 dit '1'", () => {
    // arrange
    const fizzBuzz = new FizzBuzz()
    const expected = '1'
    // act
    const actual = fizzBuzz.say(1)
    // assert
    expect(actual).to.equal(expected)
  });

  [5, 10].forEach((key) => {
    it(`${key} dit 'Buzz'`, () => {
      // arrange
      const fizzBuzz = new FizzBuzz()
      const expected = 'Buzz'
      // act
      const actual = fizzBuzz.say(key)
      // assert
      expect(actual).to.equal(expected)
    })
  });

  [3, 6].forEach((key) => {
    it(`${key} dit 'Fizz'`, () => {
      // arrange
      const fizzBuzz = new FizzBuzz()
      const expected = 'Fizz'
      // act
      const actual = fizzBuzz.say(key)
      // assert
      expect(actual).to.equal(expected)
    })
  })

  it("15 dit 'FizzBuzz'", () => {
    // arrange
    const fizzBuzz = new FizzBuzz()
    const expected = 'FizzBuzz'
    // act
    const actual = fizzBuzz.say(15)
    // assert
    expect(actual).to.equal(expected)
  });
})
